prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -270,8 +271,15 @@ class ReportFunction(
         actionHistory.trackActionResponse(response, report, workflowEngine.settings)
         workflowEngine.recordAction(actionHistory)
 
+        var bypassMessageQueue = false
+        if (options == Options.BypassQueueForQualityFilters) {
+            bypassMessageQueue = responseContainsQualityFilter(response.body.toString())
+        }
+
         // queue messages here after all task / action records are in
-        actionHistory.queueMessages(workflowEngine)
+        if (!bypassMessageQueue)
```

Comment:
This will result in a 200 being returned to the sender. Consider finding a way to return a 400 with the filter information so that it's clear that something about their request needs to be changed.

Reply:
@whytheplatypus fixed in the else block below

Reply:
@whytheplatypus actually find it ABOVE now... post-Carlos refactor merge haha